src/share/classes/com/sun/btrace/agent/TraceOutputWriter.java

```diff
@@ -149,7 +149,7 @@ private void nextWriter() {
         private FileWriter getNextWriter() throws IOException {
         	currentFileWriter.close();
         	File scriptOutputFile_renameFrom = new File(path + File.separator + baseName);
-        	File scriptOutputFile_renameTo = new File(path + File.separator + baseName + "." + (counter++));
+        	File scriptOutputFile_renameTo = new File(path + File.separator + baseName.substring(0, baseName.indexOf("-")) + ".btrace." + (counter++));
```

Comment:
Is splitting on '-' in baseName intentional? When baseName does not contain '-' this code will fail.